hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClusterListenerSupport.java

```diff
@@ -233,7 +233,12 @@ public void run() {
                             connectToCluster();
                         } catch (Exception e) {
                             logger.warning("Could not re-connect to cluster shutting down the client", e);
-                            client.getLifecycleService().shutdown();
+                            new Thread(new Runnable() {
```

Comment:
Can you also add a name to the thread? This way it become easier to determine what the purpose of this thread is when doing some debugging.